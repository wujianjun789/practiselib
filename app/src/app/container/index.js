/**
 * Created by a on 2017/6/29.
 */
import React, { Component } from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux'
import PropTypes from 'prop-types';
import '../../../public/styles/app.less';
import Card from './Card';
import UserCenter from '../../common/containers/UserCenter';
import Overlayer from '../../common/containers/Overlayer';
import {loginHandler} from '../../api/login'
import {getModule} from '../action/index'
/**
 * @param {String} title required
 * @param {String} name  required
 * @param {Array}  items required
 */
export class App extends Component{
    constructor(props){
        super(props);
        this.loginFail = this.loginFail.bind(this);
        this.loginSuccess = this.loginSuccess.bind(this);
    }

    componentWillMount(){
        console.log(localStorage)
        /*if(sessionStorage.sessionID==0||sessionStorage.sessionID==null||sessionStorage.sessionID==""){
            
            this.props.router.push('/login')
            
        }else{

            this.props.actions.loginHandler(sessionStorage.username, sessionStorage.password, this.loginSuccess, this.loginFail);

        }*/
        // if(this.props.userCenter.islogin!=1){
        //     this.props.router.push('/login')
        // }
        const {actions} = this.props;
        actions && actions.getModule();
    }

    componentDidMount() {
    }

    componentWillReceiveProps(nextProps) {



    }

    loginFail(){
        this.setState({style:{visibility: 'visible'}})
    }

    loginSuccess(){
        this.props.router.push('/')
    }

    render(){
        const {title="StarRiver", name="智慧路灯管理系统", items} = this.props;
        return (
            <div className="app">
                <div className="header clearfix">
                    <div className="header-left clearfix">
                        <span className="icon icon-logo"></span>
                        <span className="tit">{title}</span>
                    </div>
                    <UserCenter router={this.props.router}/>
                    <div className="header-right clearfix">
                        <span className="name">{name}</span>
                    </div>
                </div>
                <div className="cont">
                    <ul className="clearfix">
                        {items.map((item, index)=><li key={item.key}><Card _key={item.key} title={item.title} link={item.link}/></li>)}
                    </ul>
                </div>
                <Overlayer />
            </div>
        )
    }
}

const mapStateToProps = (state, ownProps) => {
    let app = state.app;
    return {
        title: app.title,
        name: app.name,
        items: app.items,
        userCenter:state.userCenter
    }
}

const mapDispatchToProps = (dispatch, ownProps) => {
    return {
        actions: bindActionCreators({
            getModule: getModule,
            loginHandler
        }, dispatch)
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(App);

App.propTypes = {
    title: PropTypes.string,
    name: PropTypes.string,
    items: PropTypes.arrayOf(
        PropTypes.shape({
            key: PropTypes.string.isRequired,
            title: PropTypes.string.isRequired,
            link: PropTypes.string.isRequired
        })
    ).isRequired
}