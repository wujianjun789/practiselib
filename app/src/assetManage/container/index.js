/**
 * Created by a on 2017/7/17.
 */
import React,{Component} from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';

import '../../../public/styles/assetManage-model.less';
import '../../../public/styles/assetManage-statistics.less'

import HeadBar from '../../components/HeadBar'
import SideBar from '../../components/SideBar'
import Overlayer from '../../common/containers/Overlayer'

import {getModelData, TreeData} from '../../data/assetModels'
import {treeViewInit} from '../../common/actions/treeView'

import {sideBarToggled} from '../action/index'
class AssetManageIndex extends Component{
    constructor(props){
        super(props);

        this.initTreeData = this.initTreeData.bind(this);
        this.onToggle = this.onToggle.bind(this);
    }

    componentWillMount(){
        this.mounted = true;
        getModelData(()=>{this.mounted && this.initTreeData()});
    }

    componentWillUnmount(){
        this.mounted = false;
    }

    componentDidMount(){
    }

    initTreeData(){
        this.props.actions.treeViewInit(TreeData)
    }

    onToggle(node){
        const {actions} = this.props;
        actions.sideBarToggled(node);
    }

    render(){
        let path = "";
        const {routes} = this.props;
        if(routes.length>3){
            path = routes[3].path;
        }

        return <div className={"container "+"asset-"+path}>
            <HeadBar moduleName={path=="manage"?"资产管理":"资产统计"} router={this.props.router}/>
            <SideBar onToggle={this.onToggle}/>
            {this.props.children}
            <Overlayer />
        </div>
    }
}

function mapStateToProps(state) {
    return {
        userCenter:state.userCenter
    }
}

function mapDispatchToProps(dispatch) {
    return {
        actions: bindActionCreators({
            treeViewInit: treeViewInit,
            sideBarToggled: sideBarToggled
        }, dispatch)
    }
}
export default connect(
    mapStateToProps,
    mapDispatchToProps
)(AssetManageIndex);