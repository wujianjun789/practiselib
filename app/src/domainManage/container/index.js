/**
 * Created by a on 2017/7/17.
 */
import React,{Component} from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';

import HeadBar from '../../components/HeadBar'
import SideBar from '../../components/SideBar'
import Overlayer from '../../common/containers/Overlayer'

import {TreeData} from '.././../data/domainModel'
import {treeViewInit} from '../../common/actions/treeView'
import {sideBarToggled} from '../action/index'
class DomainManageIndex extends Component{
    constructor(props){
        super(props);
        this.initTreeData = this.initTreeData.bind(this);
        this.onToggle = this.onToggle.bind(this);
    }

    componentWillMount(){
        if(sessionStorage.sessionID==0||sessionStorage.sessionID==null||sessionStorage.sessionID==""){
            this.props.router.push('/login') 
        }else{}
        this.initTreeData();
    }

    componentWillUnmount(){

    }

    componentDidMount(){
        console.log(this.props.userCenter); 
    }

    initTreeData(){
        this.props.actions.treeViewInit(TreeData);
    }

    onToggle(node){
        this.props.actions.sideBarToggled(node);
    }

    render(){
        let path = "";
        const {children} = this.props;
        if(children){
            path = children.props.route.path;
        }
        return <div className={"container "+"domain-"+path}>
            <HeadBar moduleName={/*path=="domainEdit"?*/"域管理"/*:"地图预览"*/} router={this.props.router}/>
            <SideBar onToggle={this.onToggle}/>
            {this.props.children}
            <Overlayer />
        </div>
    }
}

function mapStateToProps(state) {
    return {
        userCenter:state.userCenter
    }
}

function mapDispatchToProps(dispatch) {
    return {
        actions: bindActionCreators({
            treeViewInit: treeViewInit,
            sideBarToggled: sideBarToggled
        }, dispatch)
    }
}
export default connect(
    mapStateToProps,
    mapDispatchToProps
)(DomainManageIndex);