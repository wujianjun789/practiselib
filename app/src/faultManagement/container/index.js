import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import Table from '../../components/Table';
import Page from '../../components/Page';
import SideBarInfo from '../../components/SideBarInfo';
import Select from '../../components/Select.1';
import ConfirmPopup from '../../components/ConfirmPopup';
import Immutable from 'immutable';
import { overlayerShow, overlayerHide } from '../../common/actions/overlayer';

import Content from '../../components/Content';
import { treeViewInit } from '../../common/actions/treeView';
import { injectIntl } from 'react-intl';
import { getChildDomainList } from '../../api/domain';
import { DatePicker } from 'antd';
import moment from 'moment';
import PieChart from '../../lightManage/utils/pieChart';
import { spliceInArray } from '../../util/algorithm';
import { getFaultOrAlertList } from '../../api/asset';

export class Fault extends Component {
  constructor(props) {
    super(props);
    this.state = {
      model: '',
      collapse: false,
      infoCollapse: false,
      page: Immutable.fromJS({
        pageSize: 15,
        current: 1,
        total: 0
      }),
      data: Immutable.fromJS([]),
      domainList: {
        titleField: 'name',
        valueField: 'name',
        index: 0,
        value: '',
        options: []
      },
      levelList: {
        titleField: 'title',
        valueField: 'name',
        index: 0,
        value: 'deadly',
        options: [
          {
            title: '致命',
            name: 'deadly'
          },
          {
            title: '严重',
            name: 'seriousness'
          },
          {
            title: '一般',
            name: 'ordinary'
          },
          {
            title: '提示',
            name: 'energy'
          }
        ]
      },
      start: moment(),
      end: moment(),
      statisticalInfo: {
        deadly: 50,
        serious: 30,
        general: 10,
        hint: 10
      }
    };

    this.columns = [
      {
        id: 0,
        field: 'model',
        title: this.props.intl.formatMessage({
          id: 'sysOperation.alarm.device.model'
        })
      },
      {
        id: 1,
        field: 'name',
        title: this.props.intl.formatMessage({ id: 'app.device.name' })
      },
      {
        id: 3,
        field: 'param',
        title: this.props.intl.formatMessage({ id: 'sysOperation.alarm.param' })
      },
      {
        id: 3,
        field: 'level',
        title: this.props.intl.formatMessage({ id: 'sysOperation.alarm.level' })
      },
      {
        id: 4,
        field: 'threshold',
        title: this.props.intl.formatMessage({
          id: 'sysOperation.alarm.threshold'
        })
      },
      {
        id: 5,
        field: 'value',
        title: this.props.intl.formatMessage({
          id: 'sysOperation.alarm.test.value'
        })
      },
      {
        id: 6,
        field: 'time',
        title: this.props.intl.formatMessage({ id: 'sysOperation.fault.time' })
      }
    ];
  }

  componentWillMount() {
    this.mounted = true;
    this.setState({ model: this.props.params.fault }, this.requestSearch);
    getChildDomainList(data => {
      this.mounted && this.initDomainList(data);
    });
  }

  componentWillUnmount() {
    this.mounted = false;
  }

  componentDidUpdate() {
    const { sidebarNode } = this.props;
    if (
      sidebarNode &&
      !sidebarNode.children &&
      sidebarNode.id != this.state.model
    ) {
      this.setState({ model: sidebarNode.id }, this.requestSearch);
    }
  }

  formatIntl = formatId => {
    const { intl } = this.props;
    return intl ? intl.formatMessage({ id: formatId }) : null;
  };

  initDomainList = data => {
    let domainList = Object.assign(
      {},
      this.state.domainList,
      { index: 0 },
      { value: data.length ? data[0].name : '' },
      { options: data }
    );
    this.setState({ domainList: domainList }, this.requestSearch);
  };

  requestSearch = () => {
    let model = '';
    switch (this.state.model) {
      case 'alarm':
        model = 'alerts';
        break;
      case 'fault':
        model = 'faults';
        break;
    }
    const { domainList, levelList, start, end, page } = this.state;
    const { pageSize, current } = page.toJS();
    if (domainList.options.length) {
      const domainId = domainList.options[domainList.index]['id'];
      // const level = levelList.options[levelList.index]['name'];
      const level = levelList.index;
      const offset = pageSize * (current - 1);
      const limit = pageSize;
      getFaultOrAlertList(
        model,
        domainId,
        level,
        start,
        end,
        offset,
        limit,
        res => {
          this.setState({ data: res });
        }
      );
    }
  };

  initPageSize = data => {
    let page = this.state.page.set('total', data.count);
    this.setState({
      page: page
    });
  };

  pageChange = (current, pageSize) => {
    let page = this.state.page.set('current', current);
    this.setState(
      {
        page: page
      },
      () => {
        this.requestSearch();
      }
    );
  };

  collapseHandler = id => {
    this.setState({
      [id]: !this.state[id]
    });
  };

  selectChange = (event, key) => {
    let index = event.target.selectedIndex;
    let list = this.state[key];
    list.index = index;
    list.value = list.options[index].name;
    this.setState({ [key]: list }, () => {
      this.requestSearch();
    });
  };

  dateChange = (id, value) => {
    this.setState({ [id]: value }, () => {
      const { start, end } = this.state;
      let prompt = start && end && end.isBefore(start);
      this.setState({
        prompt: Object.assign({}, this.state.prompt, { date: prompt })
      });
    });
  };

  drawChart = ref => {
    if (ref == null) {
      this.piechart.destroy();
    } else {
      const { statisticalInfo } = this.state;
      let data = [
        statisticalInfo.deadly,
        statisticalInfo.serious,
        statisticalInfo.general,
        statisticalInfo.hint
      ];
      this.piechart = new PieChart({
        wrapper: ref,
        data: data,
        color: ['#F83D59', '#FA919C', '#FA9E17', '#FBEF35']
      });
    }
  };

  render() {
    const {
      collapse,
      infoCollapse,
      page,
      data,
      domainList,
      typeList,
      levelList,
      start,
      end,
      statisticalInfo,
      allChecked,
      checked
    } = this.state;
    return (
      <Content className={'offset-right ' + (collapse ? 'collapsed' : '')}>
        <div className="heading">
          <Select
            id="domain"
            titleField={domainList.titleField}
            valueField={domainList.valueField}
            options={domainList.options}
            value={domainList.value}
            onChange={e => {
              this.selectChange(e, 'domainList');
            }}
          />
          {/* <Select id="type" titleField={typeList.titleField} valueField={typeList.valueField}
          options={typeList.options} value={typeList.value} onChange={(e) => {this.selectChange(e, 'typeList');}}/> */}
          <Select
            id="level"
            titleField={levelList.titleField}
            valueField={levelList.valueField}
            options={levelList.options}
            value={levelList.value}
            onChange={e => {
              this.selectChange(e, 'levelList');
            }}
          />
          <div className="datePicker">
            <DatePicker
              id="startDate"
              format="YYYY/MM/DD"
              placeholder="开始日期"
              style={{ width: '106px' }}
              defaultValue={start}
              value={start}
              onChange={value => this.dateChange('start', value)}
            />
            <span>-</span>
            <DatePicker
              id="endDate"
              format="YYYY/MM/DD"
              placeholder="结束日期"
              style={{ width: '106px' }}
              defaultValue={end}
              value={end}
              onChange={value => this.dateChange('end', value)}
            />
          </div>
        </div>
        <div className="table-container">
          <Table columns={this.columns} data={data} />
          <Page
            className={'page ' + (page.get('total') == 0 ? 'hidden' : '')}
            pageSize={page.get('pageSize')}
            current={page.get('current')}
            total={page.get('total')}
            onChange={this.pageChange}
          />
        </div>
        <SideBarInfo
          collapseHandler={this.collapseHandler}
          className={infoCollapse ? 'infoCollapse ' : ''}
        >
          <div className="panel panel-default statistical-info">
            <div
              className="panel-heading"
              role="presentation"
              onClick={() => {
                !collapse && this.collapseHandler('infoCollapse');
              }}
            >
              <span className="icon_chart" />
              {this.props.intl.formatMessage({
                id: 'sysOperation.statistical.info'
              })}
              <span className="icon icon_collapse pull-right" />
            </div>
            <div className={'panel-body' + (infoCollapse ? 'collapsed' : '')}>
              <div className="left">
                <div id="alarm" className="circle" ref={this.drawChart} />
              </div>
              <div className="right">
                <div className="count deadly">
                  <div className="dot" />
                  {this.formatIntl('sysOperation.count.deadly')} :{' '}
                  {statisticalInfo.deadly
                    ? statisticalInfo.deadly
                    : this.formatIntl('sysOperation.noCount')}
                </div>
                <div className="count serious">
                  <div className="dot" />
                  {this.formatIntl('sysOperation.count.serious')} :{' '}
                  {statisticalInfo.serious
                    ? statisticalInfo.serious
                    : this.formatIntl('sysOperation.noCount')}
                </div>
                <div className="count general">
                  <div className="dot" />
                  {this.formatIntl('sysOperation.count.general')} :{' '}
                  {statisticalInfo.general
                    ? statisticalInfo.general
                    : this.formatIntl('sysOperation.noCount')}
                </div>
                <div className="count hint">
                  <div className="dot" />
                  {this.formatIntl('sysOperation.count.hint')} :{' '}
                  {statisticalInfo.hint
                    ? statisticalInfo.hint
                    : this.formatIntl('sysOperation.noCount')}
                </div>
              </div>
            </div>
          </div>
        </SideBarInfo>
      </Content>
    );
  }
}

const mapStateToProps = (state, ownProps) => {
  return {
    sidebarNode: state.systemOperation.get('sidebarNode')
  };
};

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(
    {
      treeViewInit,
      overlayerShow,
      overlayerHide
    },
    dispatch
  )
});

export default connect(mapStateToProps, mapDispatchToProps)(injectIntl(Fault));
