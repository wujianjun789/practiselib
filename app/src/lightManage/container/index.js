/**
 * Created by a on 2017/8/24.
 */
import React, {Component} from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';

import '../../../public/styles/lightManage-map.less';
import '../../../public/styles/lightManage-control.less';

import HeadBar from '../../components/HeadBar';
import SideBar from '../../components/SideBar';
import Overlayer from '../../common/containers/Overlayer';

import {getModelData, TreeData} from '.././../data/lightModel';
import {treeViewInit, onToggleById} from '../../common/actions/treeView';
// import {sideBarToggled} from '../action/index'
class lightManageIndex extends Component {
  constructor(props) {
    super(props);
    this.initTreeData = this.initTreeData.bind(this);
    this.onToggle = this.onToggle.bind(this);
  }

  componentWillMount() {
    this.mounted = true;
    getModelData(null, () => {
      this.mounted && this.initTreeData();
    });
  }

  componentWillUnmount() {
    this.mounted = false;
  }

  componentDidMount() {
  }

  componentWillReceiveProps(nextProps){
    if(nextProps.location.pathname !== this.props.location.pathname){
      this.props.actions.onToggleById(nextProps.location.pathname);
    }
  }

  initTreeData() {
    this.props.actions.treeViewInit(TreeData);
  }

  onToggle(node) {
    // console.log(node);
    // this.props.actions.sideBarToggled(node);
  }

  render() {
    let parentPath = '';
    let childPath = '';
    const {routes} = this.props;
    if (routes.length > 4) {
      parentPath = routes[4].path;
    }

    if (routes.length > 5) {
      childPath = routes[5].path;
    }
    return <div className={'container light-' + parentPath + ' ' + parentPath + '-' + childPath}>
      <HeadBar moduleName={'app.light'} router={this.props.router}/>
                    
      <SideBar onToggle={this.onToggle}/>
      {this.props.children}
      <Overlayer />
    </div>;
  }
}

function mapStateToProps(state) {
  return {
    userCenter:state.userCenter,
  };
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators({
      treeViewInit: treeViewInit,
      onToggleById: onToggleById
      // sideBarToggled: sideBarToggled
    }, dispatch),
  };
}
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(lightManageIndex);