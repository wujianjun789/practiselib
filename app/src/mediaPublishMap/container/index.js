/**
 * Created by a on 2017/10/19.
 */
import React,{Component} from 'react';

import Content from '../../components/Content';
import MapView from '../../components/MapView';
export default class MediaPublishMap extends Component{
    constructor(props){
        super(props);
        this.state = {
            map:{
                id: "mediaPublishMap",
                latlng: []
            }
        };
    }

    render(){
        const {map} = this.state;
        return <Content>
            <MapView mapData={map}></MapView>
        </Content>
    }
}