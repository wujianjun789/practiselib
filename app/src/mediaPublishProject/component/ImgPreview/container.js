/** Created By ChrisWen
 *  18/03/19
 *  ImgPreview Component
 */
import React from 'react';
import '../../../../public/styles/imgPreview.less';

const mockProps = {
  areaList: [{
    id: 1,
    style: {width: 0, height: 0, top: 0, left: 0 },
    src: '',
  }],
  selectedId: 65535,
  projectSize: { width: 0, height: 0 },
};
const { areaList: initAreaList, selectedId: initSelectedId, projectSize: initProjectSize } = mockProps;

function renderAreaList(areaList, selectedId) {
  const renderList = areaList.map((area) => {
    const { id, style } = area;
    const className = id === selectedId ? 'active' : '';
    return (
      <div className={`areaEle ${className}`} key={id} style={style}>
        <img src={area.src} alt="" style={{width:style.width, height: style.height}}/>
      </div>);
  });
  return renderList;
}

function getCorrectStyle(status, projectSize) {
  let style = {};
  switch (status) {
  case 0:
    style = { marginLeft: -0.5 * projectSize.width, marginTop: -0.5 * projectSize.height, ...projectSize };
    return style;
  case 1:
    style = { top: 0, marginLeft: -0.5 * projectSize.width, ...projectSize };
    return style;
  case 2:
    style = { left: 0, marginTop: -0.5 * projectSize.height, ...projectSize };
    return style;
  case 3:
    style = { left: 0, top: 0, ...projectSize };
    return style;  
  default:
    return style;
  }
}

const PreviewContainer = (props) => {
  /* We have reached an agreement that 65535 is the number can not be touched.
   * 65535 means null;
   */
  const { projectSize = initProjectSize, areaList = initAreaList, selectedId = initSelectedId, status = 0, ...otherProps } = props;
  const style = getCorrectStyle(status, projectSize);
  const areaListEle = renderAreaList(areaList, selectedId);
  return (
    <div className="container-wapper">
      <div id="preview-container" style={style}>{areaListEle}</div>
    </div>
  );
};


export default PreviewContainer;