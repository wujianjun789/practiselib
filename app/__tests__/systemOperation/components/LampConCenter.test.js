import React from 'react';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import {LampConCenter} from '../../../src/systemOperation/container/LampConCenter';
import Immutable from 'immutable';

describe('<LampConCenter /> component', () => {
    const lampConCenter = Immutable.fromJS({
        domain: {
            value: '',
            titleField: 'value',
            valueField: 'value',
            options: [
                {value: 'domain1'},
                {value: 'domain2'},
                {value: 'domain3'},
                {value: 'domain4'},
                {value: 'domain5'},
                {value: 'domain6'},
                {value: 'domain7'},
                {value: 'domain8'},
                {value: 'domain9'},
                {value: 'domain10'},
                {value: 'domain11'},
            ]
        },
        page: {
            pageSize: 10,
            current: 1,
            total: 21
        },
        data: [
            {id:1,deviceName: '灯集中控制器', model: '001', deviceID: '00158D0000CABAD5', model: 8080, lng: '000.000.000.000', lat: '000.000.000.000'},
            {id:2,deviceName: '灯集中控制器', model: '002', deviceID: '00158D0000CABAD5', model: 8080, lng: '000.000.000.000', lat: '000.000.000.000'},
            {id:3,deviceName: '灯集中控制器', model: '003', deviceID: '00158D0000CABAD5', model: 8080, lng: '000.000.000.000', lat: '000.000.000.000'},
            {id:4,deviceName: '灯集中控制器', model: '004', deviceID: '00158D0000CABAD5', model: 8080, lng: '000.000.000.000', lat: '000.000.000.000'},
            {id:5,deviceName: '灯集中控制器', model: '005', deviceID: '00158D0000CABAD5', model: 8080, lng: '000.000.000.000', lat: '000.000.000.000'},
            {id:6,deviceName: '灯集中控制器', model: '006', deviceID: '00158D0000CABAD5', model: 8080, lng: '000.000.000.000', lat: '000.000.000.000'},
            {id:7,deviceName: '灯集中控制器', model: '007', deviceID: '00158D0000CABAD5', model: 8080, lng: '000.000.000.000', lat: '000.000.000.000'},
            {id:8,deviceName: '灯集中控制器', model: '008', deviceID: '00158D0000CABAD5', model: 8080, lng: '000.000.000.000', lat: '000.000.000.000'},
            {id:9,deviceName: '灯集中控制器', model: '009', deviceID: '00158D0000CABAD5', model: 8080, lng: '000.000.000.000', lat: '000.000.000.000'},
            {id:10,deviceName: '灯集中控制器', model: '010', deviceID: '00158D0000CABAD5', model: 8080, lng: '000.000.000.000', lat: '000.000.000.000'},
            {id:11,deviceName: '灯集中控制器', model: '011', deviceID: '00158D0000CABAD5', model: 8080, lng: '000.000.000.000', lat: '000.000.000.000'}
        ],
        popupInfo: {
            whitelistData: [
                {id:1,name: '灯集中控制器', number: '00158D0000CABAD5', model: 8080, lng: '000.000.000.000', lat: '000.000.000.000'},
                {id:2,name: '灯集中控制器', number: '00158D0000CABAD5', model: 8080, lng: '000.000.000.000', lat: '000.000.000.000'},
                {id:3,name: '灯集中控制器', number: '00158D0000CABAD5', model: 8080, lng: '000.000.000.000', lat: '000.000.000.000'},
                {id:4,name: '灯集中控制器', number: '00158D0000CABAD5', model: 8080, lng: '000.000.000.000', lat: '000.000.000.000'},
                {id:5,name: '灯集中控制器', number: '00158D0000CABAD5', model: 8080, lng: '000.000.000.000', lat: '000.000.000.000'},
                {id:6,name: '灯集中控制器', number: '00158D0000CABAD5', model: 8080, lng: '000.000.000.000', lat: '000.000.000.000'},
                {id:7,name: '灯集中控制器', number: '00158D0000CABAD5', model: 8080, lng: '000.000.000.000', lat: '000.000.000.000'},
                {id:8,name: '灯集中控制器', number: '00158D0000CABAD5', model: 8080, lng: '000.000.000.000', lat: '000.000.000.000'},
                {id:9,name: '灯集中控制器', number: '00158D0000CABAD5', model: 8080, lng: '000.000.000.000', lat: '000.000.000.000'},
                {id:10,name: '灯集中控制器', number: '00158D0000CABAD5', model: 8080, lng: '000.000.000.000', lat: '000.000.000.000'},
                {id:11,name: '灯集中控制器', number: '00158D0000CABAD5', model: 8080, lng: '000.000.000.000', lat: '000.000.000.000'}
            ],
            domainList: {
                titleField: 'title',
                valueField: 'value',
                options: [
                    {id: 1, title: 'domain01', value: 'domain01'},
                    {id: 2, title: 'domain02', value: 'domain02'},
                    {id: 3, title: 'domain03', value: 'domain03'},
                    {id: 4, title: 'domain04', value: 'domain04'},
                    {id: 5, title: 'domain05', value: 'domain05'},
                    {id: 6, title: 'domain06', value: 'domain06'},
                    {id: 7, title: 'domain07', value: 'domain07'}
                ]
            },
            modelList: {
                titleField: 'title',
                valueField: 'value',
                options: [
                    {id: 1, title: 'model01', value: 'model01'},
                    {id: 2, title: 'model02', value: 'model02'},
                    {id: 3, title: 'model03', value: 'model03'},
                    {id: 4, title: 'model04', value: 'model04'},
                    {id: 5, title: 'model05', value: 'model05'},
                    {id: 6, title: 'model06', value: 'model06'},
                    {id: 7, title: 'model07', value: 'model07'}
                ]
            }
        }
    });
    const columns = [
        {id: 1, field: "deviceName", title: "设备名称"},
        {id:2, field: "model", title: "型号"},
        {id:3, field: "deviceID", title: "设备编号"},
        {id:4, field: "model", title: "端口号"},
        {id:5, field: "lng", title: "经度"},
        {id:6, field: "lat", title: "纬度"},
    ];
    it('render normal', () => {
        const cmp = shallow(<LampConCenter lampConCenter={lampConCenter} />);
        
        const header = cmp.find('.heading');
        const select = header.find('Select');
        const domain = lampConCenter.get('domain').toJS();
        expect(select.prop('titleField')).toBe(domain.titleField);
        expect(select.prop('valueField')).toBe(domain.valueField);
        expect(select.props().options).toEqual(domain.options);
        expect(select.props().value).toBe(domain.value);
        const searchText = header.find('SearchText');
        expect(searchText.prop('placeholder')).toBe('输入设备名称');
        expect(searchText.prop('value')).toBe('');
        const btn = header.find('#sys-add');
        expect(btn.text()).toBe('添加');

        const table = cmp.find('Table');
        expect(table.prop('columns')).toEqual(columns);
        expect(table.prop('data')).toEqual(lampConCenter.get('data'));

        const page = cmp.find('Page');
        const pageData = lampConCenter.get('page').toJS();
        expect(page.prop('pageSize')).toBe(pageData.pageSize);
        expect(page.prop('current')).toBe(pageData.current);
        expect(page.prop('total')).toBe(pageData.total);

        const sidebarInfo = cmp.find('SideBarInfo');
        expect(sidebarInfo.find('.device-statics-info').at(0).find('.panel-heading').text()).toBe('选中设备');
        expect(sidebarInfo.find('.whitelist .panel-heading').text()).toBe('白名单');
    });

    it('snapshot', () => {
        const cmp = renderer.create(<LampConCenter lampConCenter={lampConCenter} />);
        const tree = cmp.toJSON();
        expect(tree).toMatchSnapshot();
    });
})