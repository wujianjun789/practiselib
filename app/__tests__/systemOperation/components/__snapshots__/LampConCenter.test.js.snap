exports[`<LampConCenter /> component snapshot 1`] = `
<div
  className="content offset-right ">
  <div
    className="heading">
    <select
      className="select form-control "
      id="domain"
      onChange={[Function]}
      value="">
      <option
        value={undefined} />
      <option
        value={undefined} />
      <option
        value={undefined} />
      <option
        value={undefined} />
      <option
        value={undefined} />
      <option
        value={undefined} />
      <option
        value={undefined} />
    </select>
    <div
      className="searchText "
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}>
      <input
        className="form-control"
        onChange={[Function]}
        placeholder="输入设备名称"
        type="search"
        value="" />
      <ul
        className="" />
      <span
        className="glyphicon glyphicon-search"
        onClick={[Function]} />
    </div>
    <button
      className="btn btn-primary add-domain"
      id="sys-add"
      onClick={[Function]}>
      添加
    </button>
  </div>
  <div
    className="lcc">
    <div
      className="table-container">
      <div
        className="table-responsive ">
        <table
          className="table table-hover">
          <thead>
            <tr>
              <th
                className="hidden">
                <input
                  checked={undefined}
                  onChange={[Function]}
                  type="checkbox" />
              </th>
              <th>
                域
              </th>
              <th>
                设备名称
              </th>
              <th>
                型号
              </th>
              <th>
                设备编号
              </th>
              <th>
                经度
              </th>
              <th>
                纬度
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
      <ul
        className="rc-pagination page hidden"
        style={Object {}}
        unselectable="unselectable">
        <li
          aria-disabled={true}
          className="rc-pagination-disabled rc-pagination-prev"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex="0"
          title="上一页">
          <a>
            
          </a>
        </li>
        <li
          aria-disabled={true}
          className="rc-pagination-disabled rc-pagination-next"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex="0"
          title="下一页">
          <a>
            
          </a>
        </li>
        <div
          className="rc-pagination-options" />
      </ul>
    </div>
  </div>
  <div
    className="container-fluid sidebar-info ">
    <div
      className="row collapse-container"
      onClick={[Function]}>
      <span
        className="icon_verital" />
    </div>
    <div
      className="panel panel-default device-statics-info">
      <div
        className="panel-heading">
        <span
          className="icon_sys_select" />
        选中设备
      </div>
      <div
        className="panel-body domain-property">
        <span
          className="domain-name">
          
        </span>
        <button
          className="btn btn-primary pull-right"
          disabled={true}
          id="sys-update"
          onClick={[Function]}>
          编辑
        </button>
        <button
          className="btn btn-danger pull-right"
          disabled={true}
          id="sys-delete"
          onClick={[Function]}>
          删除
        </button>
      </div>
    </div>
    <div
      className="panel panel-default device-statics-info whitelist">
      <div
        className="panel-heading">
        <span
          className="icon_sys_whitelist" />
        白名单
      </div>
      <div
        className="panel-body domain-property">
        <span
          className="domain-name">
          包含：0 个项目
        </span>
        <button
          className="btn btn-primary pull-right"
          disabled={true}
          id="sys-whitelist"
          onClick={[Function]}>
          编辑
        </button>
      </div>
    </div>
    <div
      className="panel panel-default map-position">
      <div
        className="panel-heading">
        <span
          className="icon_map_position" />
        地图位置
      </div>
      <div
        className="map-container panel-body">
        <div
          className="map-view "
          id="systemOperation" />
      </div>
    </div>
  </div>
</div>
`;
